### DISCUSSION TITLE
What do folks use for synchronous discussion/drive-by questions?

### DISCUSSION THREAD

**Discussion by sanjayankur31** [OCNS/SoftwareWG#13] [General]
I know we all have mailing lists and forums for our software communities, but what do people use for synchronous discussions or one-time drive-by queries where subscribing to a mailing list or registering at a forum would be perhaps considered overkill?

Off the top of my head, the platform needs to be:

- easy to join and leave (without requiring an invite: so Slack is out)
- should ideally have read-access without registering so folks can see what's going on
- should be hosted so we don't need to find infrastructure and administer it
- free of cost if at all possible
- Free/Open Source so we keep in line with Open Science philosophies.
- integration with GitHub/GitLab notifications etc. would perhaps be nice for a -developers channel but isn't necessary.

In the past I've leaned towards [Gitter](https://gitter.im) because of its scalability and integration with GitHub, but I'm now looking at [Matrix/Element](https://element.io) which we're using more and more in the Fedora community (with plans to migrate off IRC to matrix permanently in the future). [Gitter has been bought out by Element who own and develop Matrix recently also](https://about.gitlab.com/blog/2020/09/30/gitter-moves-to-element/).

I guess a related question is: how important is such a communication channel? 

I think it's quite important, even if they're not super active in our smaller software communities. We don't have these for NeuroML/OSB yet, but we do have them for NeuroFedora where we see quite a bit of activity.

**Comment by brenthuisman** [comment]
We actually don't have mailing lists for Arbor dev. Our main method of communication is a Slack channel, to which we invite everyone who requests access. We use this for both synchronous and asynchronous communication.

Not too happy with Slack either, for the reasons you list, plus, we're freeriding, so history is limited. This is seen as a feature or a bug, depending on who you ask ;) Generally I'm a fan of Gitter, because even though Github is also not open source, it's quite open and people are virtually guaranteed to have an account already, and as you point out, Gitter itself is in good hands and accessible with Matrix clients.

As this is our main communication channel, both for the core team and with our partners, it's quite important :)

We've opened the Discussions section on our Github repo a few months ago, as that has an even lower barrier to entry. A more synchronous channel can always be set up is need be is our idea.

**Comment by stewart-heitmann** [comment]
Hmm. Gitter sounds interesting! Does it allow users to add attachments (eg source code)?

I am currently looking for something for my own project. I've been contemplating SourceForge Discussion Groups but haven’t taken the plunge because I suspect there may be better options out there these days.

Stewart Heitmann
Brain Dynamics Toolbox

**Reply by sanjayankur31** [reply]
> Hmm. Gitter sounds interesting! Does it allow users to add attachments (eg source code)?

It uses [GitHub markdown](https://docs.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax), so I think you paste your source code as you would on GitHub in an issue etc. and it formats it properly. I don't *think* it does uploads at the moment. Matrix/Element does, so one would expect Gitter to gain that functionality soon too now that they're part of the same eco system. For the moment, on Gitter, you have to use a [GitHub gist](https://gist.github.com/) or a [paste server](https://paste.fedoraproject.org) and something like [imgur](imgur.com) for screenshots/images.

Their full documentation is [here](https://gitlab.com/gitterHQ/webapp/tree/develop/docs#documentation).

**Comment by stewart-heitmann** [comment]
Lovely! I’ll check it out. Thank you Ankur.

- Stewart

**Reply by sanjayankur31** [reply]
I've just noticed that Gitter rooms are *already* bridged to Matrix. So, for example, this is the [Gitter room for Brian2](https://gitter.im/brian-team/brian2), and it has a [link to use it over Matrix](https://matrix.to/#/%23brian-team_brian2:gitter.im?utm_source=gitter). You can use either or even [Gitter IRC](https://irc.gitter.im/), and they're all bridged.
